### PR TITLE
refactor(demo): update navigation and demo

### PR DESCRIPTION
**Pull request by CITguy**
(a.k.a. Demo Reorganization - part 2)

* Demo navigation now follows layout expected from Design
* lots of reorganization of demo code
  * many examples added/moved
  * many templates added/moved
  * `demoApp` now only includes `.config()` and `.run()`
    * additional logic moved to appropriate locations
* component:progressbar moved to Molecules
* component:grids docs removed and content merged with layout:grids
* `encore.ui.molecules` depends on `encore.ui.quarks` and `encore.ui.atoms`
* `encore.ui.atoms` depends on `encore.ui.quarks`
* modules in search and nav now sorted by their display name
  * allows easier organization later on
* update TravisCI to use node 4.2.3 LTS
* support `.node-version` locally
* removed unnecessary documentation
  * Overview > EncoreUI Design section
  * Styleguide > Basics page

### LGTMs
- [ ] Dev LGTM
- [ ] Dev LGTM
- [x] Design LGTM

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 sudo: false
 language: node_js
 node_js:
-- '0.12'
+  - "4.2.3" # same as .node-version
```
Comment (CITguy) [inline .travis.yml]: Updated to latest NodeJS LTS version

File: demo/examples/layout.grids.html [hunk 2]
```diff
@@ -0,0 +1,96 @@
+<h3>Two u-1-2 modules</h3>
```
Comment (CITguy) [inline demo/examples/layout.grids.html]: This file will eventually go away with removal of pure CSS

File: demo/less/_rxExample.less [hunk 3]
```diff
@@ -1,4 +1,15 @@
...
+  h1,
+  h2,
+  h3,
+  h4,
+  h5,
+  h6 {
+    &:first-of-type {
+      margin-top: 0;
+    }
+  }
```
Comment (CITguy) [inline demo/less/_rxExample.less]: heading styles were scoped to `.component-demo`, but that style is no longer used

File: demo/less/demoApp.less [hunk 4]
```diff
@@ -66,18 +66,7 @@
   .readme-title {
     margin-top: 0;
   }
-}//.component-description
-
-.styleguide-character-count {
```
Comment (CITguy) [inline demo/less/demoApp.less]: unused class

File: demo/less/demoApp.less [hunk 5]
```diff
@@ -149,30 +138,6 @@ p {
...
-  h4,
-  h5,
-  h6 {
-    &:first-child {
-      margin-top: 0;
-    }
-  }
-}
-
-rx-page.styleguide {
```
Comment (CITguy) [inline demo/less/demoApp.less]: no longer used

File: demo/scripts/demoApp.js [hunk 6]
```diff
@@ -463,133 +384,4 @@ angular.module('demoApp', ['encore.ui', 'ngRoute'])
             $window.scrollTo(0,0);
         }
     });
-})
-
-// Used by the Tables styleguide page to show pagination
-.controller('tablePaginateExampleController', function ($scope, PageTracking) {
```
Comment (CITguy) [inline demo/scripts/demoApp.js]: moved to `demo/examples/table.paginate.js`

File: demo/scripts/demoApp.js [hunk 7]
```diff
@@ -463,133 +384,4 @@ angular.module('demoApp', ['encore.ui', 'ngRoute'])
...
-    $scope.pager = PageTracking.createInstance();
-    $scope.people = [
-        { name: 'Patrick Deuley', occupation: 'Design Chaplain' },
-        { name: 'Hussam Dawood', occupation: 'Cat Lover' }
-    ];
-})
-
-// Used by the Tables styleguide page to show nested content
-.controller('tableNestedExampleCtrl', function ($scope) {
```
Comment (CITguy) [inline demo/scripts/demoApp.js]: moved to `demo/examples/table.nestedTable.js`

File: demo/scripts/demoApp.js [hunk 8]
```diff
@@ -463,133 +384,4 @@ angular.module('demoApp', ['encore.ui', 'ngRoute'])
...
-            { name: 'Spike', animal: 'Porcupine', age: 10 }
-        ] },
-        { name: 'Hussam Dawood', occupation: 'Cat Lover', pets: [
-            { name: 'Sassy', animal: 'Cat', age: 6 }
-        ] }
-    ];
-})
-
-// Used by the Tables styleguid page to show table with filtering capabilities
-.controller('tableWithFilteringCtrl', function ($scope, PageTracking) {
```
Comment (CITguy) [inline demo/scripts/demoApp.js]: moved to `demo/examples/table.filtering.js`

File: demo/scripts/demoApp.js [hunk 9]
```diff
@@ -463,133 +384,4 @@ angular.module('demoApp', ['encore.ui', 'ngRoute'])
...
-        { name: 'Hussam Dawood', occupation: 'Cat Lover' },
-        { name: 'Kevin Lamping', occupation: 'Framework Father' },
-        { name: 'Glynnis Ritchie', occupation: 'Serif Sheriff' },
-        { name: 'Freddy Knuth', occupation: 'Venezuelan Hurricane' },
-        { name: 'Chris Cantu', occupation: 'Texan Tornado' },
-    ];
-})
-
-// Used by the Layout 2 styleguide page to show pagination
-.controller('layout2StyleguideCtrl', function ($scope, PageTracking) {
```
Comment (CITguy) [inline demo/scripts/demoApp.js]: moved to `demo/scripts/layout/page/dataTableController.js`

File: demo/scripts/demoApp.js [hunk 10]
```diff
@@ -463,133 +384,4 @@ angular.module('demoApp', ['encore.ui', 'ngRoute'])
...
-        { name: 'Hussam Dawood', occupation: 'Cat Lover', number: 2, status: 'DISABLED' },
-        { name: 'Kevin Lamping', occupation: 'Framework Father', number: 3, status: 'ERROR' },
-        { name: 'Glynnis Ritchie', occupation: 'Serif Sheriff', number: 4, status: 'INFO' },
-        { name: 'Freddy Knuth', occupation: 'Venezuelan Hurricane', number: 5, status: 'WARNING' },
-        { name: 'Chris Cantu', occupation: 'Texan Tornado', number: 6, status: 'PENDING' },
-    ];
-})
-
-// Used by the Customizing Buttons styleguide page
-.controller('customButtonsStyleguideCtrl', function ($scope, $timeout) {
```
Comment (CITguy) [inline demo/scripts/demoApp.js]: moved to `demo/examples/button.animated.js`

File: demo/scripts/demoApp.js [hunk 11]
```diff
@@ -463,133 +384,4 @@ angular.module('demoApp', ['encore.ui', 'ngRoute'])
...
-    };
-
-    $scope.clickMe = function () {
-        $scope.status.loading = true;
-        $timeout(function () {
-            $scope.status.loading = false;
-        }, 4000);
-    };
-})
-.controller('commentBoxStyleguideCtrl', function ($scope) {
```
Comment (CITguy) [inline demo/scripts/demoApp.js]: unused, removed

File: demo/scripts/demoApp.js [hunk 12]
```diff
@@ -463,133 +384,4 @@ angular.module('demoApp', ['encore.ui', 'ngRoute'])
...
-    $scope.comment = '';
-})
-
-// Used by the Filters styleguide page to show a collapsible region
-.controller('collapseFilterExampleCtrl', function ($scope) {
```
Comment (CITguy) [inline demo/scripts/demoApp.js]: moved to `demo/examples/table.filteringCollapsible.js`

File: demo/scripts/demoApp.js [hunk 13]
```diff
@@ -463,133 +384,4 @@ angular.module('demoApp', ['encore.ui', 'ngRoute'])
...
-        { name: 'General1-1', ram: '1 GB', cpu: 1, disk: '20GB SSD', region: 'DFW'},
-        { name: 'General1-2', ram: '2 GB', cpu: 2, disk: '40GB SSD', region: 'ORD'},
-        { name: 'General1-4', ram: '4 GB', cpu: 4, disk: '80GB SSD', region: 'IAD'},
-        { name: 'General1-8', ram: '8 GB', cpu: 8, disk: '160GB SSD', region: 'LON'},
-        { name: 'I/O1-15', ram: '15 GB', cpu: 4, disk: '40GB SSD', region: 'HKG'},
-        { name: 'I/O1-30', ram: '30 GB', cpu: 8, disk: '40GB SSD', region: 'SYD'}
-    ];
-})
-
-.controller('formsDisabledExamplesCtrl', function ($scope) {
```
Comment (CITguy) [inline demo/scripts/demoApp.js]: moved to `demo/examples/forms.disabled.js`

File: demo/scripts/demoApp.js [hunk 14]
```diff
@@ -463,133 +384,4 @@ angular.module('demoApp', ['encore.ui', 'ngRoute'])
...
-    $scope.selDisabled = 'disabled';
-    $scope.radDisabled = 1;
-    $scope.chkDisabledOne = true;
-    $scope.chkDisabledTwo = false;
-    $scope.togDisabledOn = true;
-    $scope.togDisabledOff = false;
-    $scope.txtAreaDisabled = 'Disabled Textarea';
-})
-
-.controller('formsInvalidExamplesCtrl', function ($scope) {
```
Comment (CITguy) [inline demo/scripts/demoApp.js]: moved to `demo/examples/forms.validation.js`

File: demo/scripts/demoApp.js [hunk 15]
```diff
@@ -463,133 +384,4 @@ angular.module('demoApp', ['encore.ui', 'ngRoute'])
...
-    $scope.selInvalid = 'invalid';
-    $scope.radInvalid = 1;
-    $scope.chkInvalidOne = true;
-    $scope.chkInvalidTwo = false;
-    $scope.togInvalidOn = true;
-    $scope.togInvalidOff = false;
-    $scope.txtAreaInvalid = 'Invalid Value';
-})
-
-.controller('formsAutoSaveExampleController', function ($scope, rxAutoSave) {
```
Comment (CITguy) [inline demo/scripts/demoApp.js]: moved to `demo/examples/forms.autoSave.js`

File: demo/scripts/demoApp.js [hunk 16]
```diff
@@ -463,133 +384,4 @@ angular.module('demoApp', ['encore.ui', 'ngRoute'])
...
-    $scope.forms = { autosave: '' };
-    rxAutoSave($scope, 'forms');
-})
-
-.controller('formsManualSaveExampleController', function ($scope, $timeout, rxNotify) {
```
Comment (CITguy) [inline demo/scripts/demoApp.js]: moved to `demo/examples/forms.manualSave.js`

**Comment by glynnis** [comment]
Design LGTM :+1: :clap: :clap:

**Comment by glynnis** [comment]
@CITguy Does this need to be run against upstream so screenshots can be merged for visual regression?

**Comment by CITguy** [comment]
@glynnis, it shouldn't need to, but I'll resubmit to check visual regression, just in case.

**Comment by CITguy** [comment]
Closing, resubmitting via upstream branch

**Pull request closed without merging**